Sources/JavaScriptKit/BasicObjects/JSTypedArray.swift

```diff
@@ -9,37 +9,40 @@ public protocol TypedArrayElement: JSValueConvertible, JSValueConstructible {
     static var typedArrayClass: JSFunctionRef { get }
 }
 
-public class JSTypedArray<Element>: JSObjectRef, ExpressibleByArrayLiteral where Element: TypedArrayElement {
+public class JSTypedArray<Element>: JSValueConvertible, ExpressibleByArrayLiteral where Element: TypedArrayElement {
+    let ref: JSObject
+    public func jsValue() -> JSValue {
+        .object(ref)
+    }
+
     public subscript(_ index: Int) -> Element {
         get {
-            return Element.construct(from: getJSValue(this: self, index: Int32(index)))!
+            return Element.construct(from: getJSValue(this: ref, index: Int32(index)))!
         }
         set {
-            setJSValue(this: self, index: Int32(index), value: newValue.jsValue())
+            setJSValue(this: ref, index: Int32(index), value: newValue.jsValue())
         }
     }
+    
+    public init(_ object: JSObject) {
```

Comment:
Does this need to be pubic? I worry that someone would pass a non-TypedArray in.

Reply:
I added a failable init instead of this. What do you think about the new one?